lib/ansible/cli/__init__.py

```diff
@@ -520,19 +519,7 @@ def _play_prereqs():
 
         # create the variable manager, which will be shared throughout
         # the code, ensuring a consistent view of global variables
-        variable_manager = VariableManager(loader=loader, inventory=inventory)
-
-        # If the basedir is specified as the empty string then it results in cwd being used.  This
-        # is not a safe location to load vars from
-        if options.get('basedir', False) is not False:
-            if basedir:
-                variable_manager.safe_basedir = True
-        else:
-            variable_manager.safe_basedir = True
-
-        # load vars from cli options
-        variable_manager.extra_vars = load_extra_vars(loader=loader)
-        variable_manager.options_vars = load_options_vars(CLI.version_info(gitinfo=False))
+        variable_manager = VariableManager(loader=loader, inventory=inventory, cli=CLI)
```

Comment:
I still think we could do a little better, likely be extracting common functionality out of `ansible.cli.CLI.version_info` into `ansible.utils`

The only reason we are passing the CLI is for that info, and the `VariableManager` doesn't need anything else.

Decomposing the code more is what I would recommend here, to make it more flexible, and without creating as many inter-object dependencies where it isn't actually needed.